support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+                        return
+                    else
+                        setBldrUrl
+                    fi
+        fi
+    fi
+}
+
+credSelect() {
+    echo ""
+    read -r -p "Would you like to use these credentials? [y/N] " response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        return
+    else
+        echo ""
+        read -r -p "Would you like to configure with custom credentials now? [y/N]" response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please reconfigure your AWS credentials and re-run s3migrate."
+            exit 1
+        fi
+    fi
+}
+
+credCheck(){
+    CREDS=( "${HOME}/.aws/credentials" "${HOME}/.aws/config" "/root/.aws/credentials" "/root/.aws/config")
+    for location in "${CREDS[@]}"; do
+        if [ -f "${location}" ]; then
+            echo ""
+            echo "AWS Credentials file located at ${location}"
+            cat "${location}"
+            credSelect
+            credsConfigured=true
+            break
+        fi
+
+        if [[ -n ${AWS_ACCESS_KEY_ID:-} ]]; then
+            if [[ -n ${AWS_SECRET_ACCESS_KEY:-} ]]; then
+                echo ""
+                echo "AWS Credentials configured via ENVVAR detected."
+                echo ""
+                echo "aws_access_key_id=${AWS_ACCESS_KEY_ID:-}"
+                echo "aws_secret_access_key=${AWS_SECRET_ACCESS_KEY:-}"
+                credSelect
+                credsConfigured=true
+                break
+            else
+                echo ""
+                echo "WARNING: Incomplete AWS Credentials configured via ENVVAR."
+                echo "Make sure to set AWS_ACCESS_KEY_ID && AWS_SECRET_ACCESS_KEY"
+                break
+            fi
+        fi
+    done
+
+    if  "${credsConfigured}" ; then
+        echo ""
+        echo "Credentials configured!"
+    else
+        echo ""
+        echo "WARNING: No AWS credentials detected!"
+        read -r -p "Would you like to generate them now? [y/N] " response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please configure your AWS credentials and re-run s3migrate."
+            echo ""
+            exit 1
+        fi
+    fi
+}
+
+welcome() {
+    echo ""
+    echo "==========================================================="
+    echo "###########################################################"
+    echo "############## Bldr Build Dep Ingest/Migrate ##############"
+    echo "###########################################################"
+    echo "==========================================================="
+    echo ""
+    echo "This tool will scrape all the hart files from an s3 or minio"
+    echo "bucket and reingest them to populate build dep and tdep metadata"
+    credsConfigured=false
+
+    setBldrUrl
+
+
+
+    if [[ -n "${HAB_AUTH_TOKEN+x}" ]]; then
```

Comment:
This seems strange. The condition is true if `HAB_AUTH_TOKEN` is set to any value, including the empty string. I'd expect the test to be for a non-empty value:
```suggestion
    if [[ -n "${HAB_AUTH_TOKEN:-}" ]]; then
```